src/mlpack/methods/preprocess/preprocess_string_util.cpp

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file preprocess_string_util.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to encode string dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include "mlpack/methods/preprocess/preprocess_string_util.hpp"
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Function neccessary to create a vector<vector<string>> by readin
+ * the contents of a file.
+ *
+ * @param filename Name of the file whose contents need to be preproccessed.
+ * @param columnDelimiter Delimiter used to split the columns of file.
+ */
+vector<vector<string>> CreateDataset(const string& filename,
+                                     char columnDelimiter)
+{
+  vector<vector<string>> dataset;
+  // Extracting the contents of file.
+  // File stream.
+  ifstream fin(filename);
+  if (!fin.is_open())
+    Log::Fatal << "Unable to open input file" << endl;
+  string line, word;
+  while (getline(fin, line))
+  {
+    stringstream streamLine(line);
+    dataset.emplace_back();
+    while (getline(streamLine, word, columnDelimiter))
+    {
+      dataset.back().push_back(word);
+    }
```

Comment:
Will this handle quoted strings in a CSV?  I know that is a tricky task but I would expect many users to have CSVs like that...

Reply:
You're right. I should definitely implement it.

Reply:
I think maybe `data::LoadCSV()` could be used for this---I believe it already handles escaped commas.  (`src/mlpack/core/data/load_csv.cpp`)